modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/legacy/session.ts

```diff
@@ -238,27 +230,28 @@ export class Session {
 		return banner;
 	}
 
-	private _setTimestamp() {
-		this._timestamp = Date.now().toString();
-
-		this._initTimestamp = this._timestamp;
+	private _getTimestamp() {
+		return parseInt(
+			Liferay.Util.LocalStorage.getItem(
+				this._timestampKey,
+				Liferay.Util.LocalStorage.TYPES.NECESSARY
+			)!,
+			10
+		);
+	}
 
-		if (navigator.cookieEnabled) {
-			Liferay.Util.Cookie.set(
-				this._cookieKey,
-				this._timestamp,
-				Liferay.Util.Cookie.TYPES.NECESSARY,
-				this._cookieOptions
-			);
-		}
+	private _setTimestamp() {
+		Liferay?.Util?.LocalStorage?.setItem(
+			this._timestampKey,
+			Date.now().toString(),
+			Liferay?.Util?.LocalStorage?.TYPES?.NECESSARY
```

Comment:
Any reason for all optional chaining? I don't see similar checks on `_getTimestamp`.

Reply:
No reason for it actually